### PR TITLE
Cleaning array

### PR DESCRIPTION
**Pull request by tbaranes**
Follow up of #51 which I did open a bit too early, it's now ready to be reviewed!
Related to #52 

This PR a cleanup of `Array` extensions:

- Removing a few methods (that I think are not that useful)
- Cleaning the code a bit
- Adding a few more utils methods
- A few improvements for the README 

@Sweeper777 As said in #51, let me know if you think there's still rooms for improvement in the current implementation, or if you think that some of the removed methods should stay.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,100 +84,102 @@ Check out the repository to find examples / tests for each feature.
...
-var myArray = [1, 2, 3]
-print(myArray.contains(array: [1, 2])) // true
-print(myArray.contains(array: [5])) // false
+var array = [0, 0, 1, 1, 2]
+array.removeAll(0)
+print(array) // [1,1,2]
 ```
 
-Get an object at a specified index:
+Check if an array contains another array:
```
Comment (Sweeper777) [inline README.md]: I think "Check if an array contains another array" sounds like checking whether an array is a sub sequence of another array, but what the `contains` method really does is checking whether an array is a subset of another array. What I mean is `[1, 2, 3]` is a subset of `[3, 2, 1, 4, 5]` but not a sub sequence. Maybe change the description to "Check if an array is a subset of another array" or "Check if an array has all the elements in another array".

File: README.md [hunk 2]
```diff
@@ -84,100 +84,102 @@ Check out the repository to find examples / tests for each feature.
...
 
 ``` swift
-var myArray = [1, 2, 3]
-print(myArray.get(index: 1)) // 2
+var array = [1, 2, 3]
+print(array.contains(array: [1, 2])) // true
+print(array.contains(array: [5])) // false
```
Comment (Sweeper777) [inline README.md]: The `contains` method does not require an argument label `array:`, right?

File: Source/ArrayExtension.swift [hunk 3]
```diff
@@ -33,52 +33,60 @@ public extension Array where Element : Equatable {
         }
     }
 
+    public mutating func removeDuplicates() {
```
Comment (Sweeper777) [inline Source/ArrayExtension.swift:36]: What about adding a non-mutating version of `removeDuplicates` and `removeAll`?

File: README.md [hunk 4]
```diff
@@ -84,100 +84,102 @@ Check out the repository to find examples / tests for each feature.
 Safely access to an element:
 
 ```swift
-var myArray = [1, 2, 3]
-print(myArray[safe: 0]) // Optional(1)
-print(myArray[safe: 10]) // nil
+var array = [1, 2, 3]
+print(array[safe: 0]) // Optional(1)
+print(array[safe: 10]) // nil
```
Comment (Sweeper777) [inline README.md:89]: Where is the safe subscript defined? I can't seem to find it.

**Review by Sweeper777 — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (tbaranes) [reply README.md:89]: First method in the Array extension: https://github.com/tbaranes/SwiftyUtils/blob/67e22067971a4c10084affe1302dd8549a2d0d09/Source/ArrayExtension.swift#L12

Comment (Sweeper777) [reply README.md:89]: Ok, I see!

**Comment by tbaranes** [comment]
All done @Sweeper777 ! Thanks for the comments 👍

**Review by Sweeper777 — APPROVED** [review]

**Pull request merged**